src/envoy/http/error_translator/filter.h

```diff
@@ -43,22 +46,24 @@ class Filter : public Envoy::Http::PassThroughEncoderFilter,
 
   bool isEspv2FilterError();
 
-  std::string errorToJson(google::rpc::Status& error);
+  Envoy::ProtobufUtil::Status errorToJson(google::rpc::Status& error,
+                                          std::string* json_out);
 
  private:
   const FilterConfigSharedPtr config_;
 
   // Current response is gRPC. Filter does not try to create a gRPC body.
   bool is_grpc_response_;
 
-  // Stores the current error that is being translated.
-  google::rpc::Status error_;
-
-  // Caches the error converted to JSON.
-  std::string error_json_;
-
   // Save the response headers so they can be modified in `encodeData`.
   Envoy::Http::ResponseHeaderMap* headers_;
+
+  // Required for resolving error details (protobuf.Any types).
+  Envoy::Protobuf::DescriptorPool descriptor_pool_;
```

Comment:
pool is just like a cache.  It make sense to store it in a global config so the cache can be shared. 
Here if we store it in filter, it could not be shared.